File: algorithm/src/main/java/baekjoon/p02292/Main.java
package baekjoon.p02292;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt()));
    }

    private static int solve(int n) {
        int cnt = 0;
        int next = 1;
        int add = 1;

        while (n >= next) {
            next += add;
            cnt++;
            add = cnt * 6;
        }

        return cnt;
    }

}

File: algorithm/src/main/java/baekjoon/p02292/Main.kt
package baekjoon.p02292

fun main(args: Array<String>) {
    println(solve(readLine()!!.toInt()))
}

private fun solve(n: Int): Int {
    var cnt = 1
    var last = 1

    while (n > last) {
        last += (cnt * 6)
        cnt++
    }

    return cnt
}